src/averageAges.js

```diff
@@ -15,13 +15,21 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mens = people.filter(person => person.sex === 'm');
+
+  if (century !== undefined) {
+    mens = mens.filter(person => Math.ceil(person.died / 100) === century);
+  }
```

Comment:
The original condition check is correct. However, the problem lies in the fact that `mens` is a subset of `people`, so when you filter `mens` based on century, you're not considering the original dataset. This means you're calculating the average age of men in a specific century out of the men who were already alive in that century, which is incorrect. You need to filter based on the entire `people` dataset, not just `mens`.